### PR TITLE
TINY-10013: Alert icon in the alert banner is only clickable when url is specified

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: TINY-10013

Description of Changes:
* Update bridge so the alert icon in the alert banner component is only clickable when `url` is specified.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/headless/components/alertbanner/AlertBannerTest.ts [hunk 1]
```diff
@@ -15,44 +15,79 @@ describe('headless.tinymce.themes.silver.components.alertbanner.AlertBannerTest'
     })
   };
 
-  const hook = TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
+  const render = (spec: any) => TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/test/ts/headless/components/alertbanner/AlertBannerTest.ts:18]:
```suggestion
  const render = (spec: Record<string, any>) => TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
```

File: modules/bridge/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Changed
+- Alert icon in alertbanner component is not clickable when `url` is not provided. #TINY-10013
```
Comment (ltrouton) [inline modules/bridge/CHANGELOG.md:10]: I believe we need an entry in the tinymce changelog.md file as well

File: modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts [hunk 3]
```diff
@@ -26,27 +28,26 @@ export const renderAlertBanner = (spec: AlertBannerWrapper, providersBackstage:
       {
         dom: {
           tag: 'div',
-          classes: [ 'tox-notification__icon' ]
+          classes: [ 'tox-notification__icon' ],
+          innerHtml: !spec.url ? icon : undefined
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts:32]: Just wanted to sanity check that we don't the standalone icon to have a tooltip?

File: modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts [hunk 4]
```diff
@@ -26,27 +28,26 @@ export const renderAlertBanner = (spec: AlertBannerWrapper, providersBackstage:
...
-            action: (comp) => {
-              AlloyTriggers.emitWith(comp, formActionEvent, { name: 'alert-banner', value: spec.url });
-            },
+            action: (comp) => AlloyTriggers.emitWith(comp, formActionEvent, { name: 'alert-banner', value: spec.url }),
             buttonBehaviours: Behaviour.derive([
               Icons.addFocusableBehaviour()
             ])
           })
-        ]
+        ] : []
```
Comment (MitchCTiny) [inline modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts:50]: Can components be undefined rather than an empty array?

**Review by MitchCTiny — APPROVED** [review]

File: modules/bridge/CHANGELOG.md (hunk 2 above)
Comment (tiny-ben-tran) [reply modules/bridge/CHANGELOG.md:10]: Added!

File: modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts (hunk 3 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts:32]: I played around with `Icons.render()` to replace `Icon.get()` and noticed no difference in terms of behaviour and tests still pass except for AlertBannerTest of course. The difference is `Icons.render()` returns the icon wrapped inside an element, which is specified in one of the arguments. I don't have a strong opinion on which should be used.

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts (hunk 3 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts:32]: I created a follow-up ticket for investigation later so we don't block this PR being merged for the coming release

File: modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts (hunk 4 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts]: Fixed

**Review by shanmen-tiny — APPROVED** [review]

**Pull request merged**